### PR TITLE
Allow juju bins to be built with the race detector enabled.

### PR DESCRIPTION
**Pull request by hpidcock**
Allows juju bins to be built with the race detector enabled for runtime race dectection.

## QA steps

- `make install COMPILE_FLAGS=-race`

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

**Review by anvial — APPROVED** [review]
QA passed.  LG2M

File: Makefile [hunk 1]
```diff
@@ -194,6 +229,10 @@ endef
 # arch of ppc64le. This canned command will do a last minute transformation of
 # the string we build the "correct" go architecture. However the build result
 # will still be placed at the expected location with names matching ppc64el.
+
+ifeq ($(FORCE_CGO),1)
+run_go_build = $(run_cgo_build)
+else
```
Comment (SimonRichardson) [inline Makefile:235]: I don't get this, why would we want to force CGO built with musl on anything but jujud?

**Comment by SimonRichardson** [comment]
@hpidcock do we still want to land this one?

**Comment by hpidcock** [comment]
Happy to revisit this in the future, as it is incredibly difficult to trigger the race detector outside of unit tests as juju just doesn't move fast enough.

**Pull request closed without merging**